esp_encrypted_img/Kconfig

```diff
@@ -0,0 +1,27 @@
+menu "Pre Encrypted OTA Configuration"
+
+    choice PRE_ENCRYPTED_OTA_SCHEME
+        prompt "Pre-encrypted OTA Scheme"
+        default PRE_ENCRYPTED_OTA_USE_RSA
+        help
+            Select the cryptographic scheme to use for pre-encrypted Over-The-Air updates.
+
+        config PRE_ENCRYPTED_OTA_USE_RSA
+            bool "RSA"
+            help
+                Use RSA for encrypting the GCM key.
+                The device will decrypt the GCM key using its private RSA key.
+
+        config PRE_ENCRYPTED_OTA_USE_ECC
+            bool "ECC"
+            help
+                Use Elliptic Curve Cryptography (ECC) for key agreement.
+                The GCM key will be derived using ECDH with a server public key
+                and a device private key (potentially derived via HMAC).
+
+    endchoice
+    
+    config PRE_ENC_OTA_HMAC_EFUSE_KEY_ID
+        int "HMAC EFUSE KEY ID"
+        default 2
```

Comment:
Alternatively, we can enforce user to set it explicitly

```
     range -1 5
     default -1
``` 
And then `static_assert` in code to ensure that value is within 0 ...5 range. Please see how its handled for `CONFIG_NVS_SEC_HMAC_EFUSE_KEY_ID` case in IDF.